de-lib/src/main/java/org/iplantc/de/apps/client/views/toolBar/AppsViewToolbarImpl.java

```diff
@@ -373,12 +373,12 @@ protected void onEnsureDebugId(String baseID) {
     }
 
     boolean containsShareableApps(List<App> apps) {
-        if(apps!=null && apps.size() > 0) {
+        if (apps != null && apps.size() > 0) {
             for (App a : apps) {
-                System.out.println(a.getAppType());
-                System.out.println(a.getPermission());
-                if (!(a.getPermission() != null && a.getPermission().equals(PermissionValue.own)
-                      && a.getAppType() != null && !a.getAppType().equals(App.EXTERNAL_APP))) {
+                if (a.getAppType().equalsIgnoreCase(App.EXTERNAL_APP) && a.isPublic()) {
+                    return false;
+                } else if (!(a.getAppType().equalsIgnoreCase(App.EXTERNAL_APP)) && !(
+                        a.getPermission() != null && a.getPermission().equals(PermissionValue.own))) {
```

Comment:
I think this would be more readable if `a.getAppType().equalsIgnoreCase(App.EXTERNAL_APP)` was refactored into a boolean var, and `a.getPermission() != null && a.getPermission().equals(PermissionValue.own)` can be condensed into `PermissionValue.own.equals(a.getPermission())`.